elide-spring/elide-spring-boot-autoconfigure/src/main/java/com/yahoo/elide/spring/controllers/ApiDocsController.java

```diff
@@ -176,4 +190,34 @@ public ResponseEntity<String> call() throws Exception {
             }
         };
     }
+
+    private String getPath(HttpServletRequest request, String prefix) {
+        String pathname = (String) request
+                .getAttribute(HandlerMapping.PATH_WITHIN_HANDLER_MAPPING_ATTRIBUTE);
+
+        pathname = pathname.replaceFirst(prefix, "");
+        try {
+            return URLDecoder.decode(pathname, StandardCharsets.UTF_8.toString());
+        } catch (UnsupportedEncodingException e) {
+            return pathname;
+        }
+    }
+
+    protected String getBaseUrl(String prefix) {
+        String baseUrl = elideConfigProperties.getBaseUrl();
+
+        if (StringUtils.isEmpty(baseUrl)) {
+            baseUrl = ServletUriComponentsBuilder.fromCurrentContextPath().build().toUriString();
+        }
+
+        if (prefix.length() > 1) {
+            if (baseUrl.endsWith("/")) {
+                baseUrl = baseUrl.substring(0, baseUrl.length() - 1) + prefix;
+            } else {
+                baseUrl = baseUrl + prefix;
```

Comment:
I'm not quite following these two branches.  Can you add a comment what you expect to be parsing here and the difference between prefix and baseUrl?

Reply:
Have added the comment. The first branch is just to remove trailing / if present on the base url. 

The base url here is the application base url which can be specified using `elide.base-url`. The prefix is configured using `elide.api-docs.path.` The method is to determine the base url of the api docs controller eg. http://www.example.org/api-docs where the application base url is `http://www.example.org` and the path for api docs is `/api-docs`.

Reply:
👍